AssemblyProvider.cs

```diff
@@ -0,0 +1,274 @@
+        switch (_signer) {
+            case Signer.CodeSign:
+                SignWithCodeSign(filePath);
+                break;
+            case Signer.Quill:
+                SignWithQuill(filePath);
+                break;
+        }
+    }
+
+    enum Signer {
+
+        CodeSign,
+        Quill
+
+    }
+
+    private static void SignWithCodeSign(string filePath) {
+        using var strip = Process.Start(new ProcessStartInfo {
+            FileName = "codesign",
+            Arguments = "--remove-signature \"" + Path.GetFullPath(filePath) + "\"",
+            UseShellExecute = false,
+            RedirectStandardError = false,
+            RedirectStandardOutput = false,
+        });
+        if (strip == null) throw new Exception("Failed to start codesign process.");
+        strip.WaitForExit();
+        if (strip.ExitCode != 0) throw new Exception("Process did not exit with status code 0. Got: " + strip.ExitCode);
+
+        using var sign = Process.Start(new ProcessStartInfo {
+            FileName = "codesign",
+            Arguments = "--force --deep --sign - \"" + Path.GetFullPath(filePath) + "\"",
+            UseShellExecute = false,
+            RedirectStandardError = false,
+            RedirectStandardOutput = false,
+        });
+        if (sign == null) throw new Exception("Failed to start codesign process.");
+        sign.WaitForExit();
+        if (sign.ExitCode != 0) throw new Exception("Process did not exit with status code 0. Got: " + sign.ExitCode);
+    }
+
+    private static void SignWithQuill(string filePath) {
+        using var strip = Process.Start(new ProcessStartInfo {
+            FileName = "quill",
+            Arguments = "sign \"" + Path.GetFullPath(filePath) + "\"",
+            UseShellExecute = false,
+            RedirectStandardError = false,
+            RedirectStandardOutput = false,
+        });
+        if (strip == null) throw new Exception("Failed to start codesign process.");
+        strip.WaitForExit();
+        if (strip.ExitCode != 0) throw new Exception("Process did not exit with status code 0. Got: " + strip.ExitCode);
+    }
+
+    private static Signer? FindSignerProgram() {
+        // Prefer quill if available, even on mac, as it has prettier output.
+        if (DoesProgramExist("quill", "sign --help", p => p.ExitCode == 0)) {
+            return Signer.Quill;
+        }
+        
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) && DoesProgramExist("codesign", "--help", p=> p.StandardError.ReadToEnd().Contains("Usage: codesign"))) {
+            return Signer.CodeSign;
+        }
+
+        return null;
+    }
+
+    private static bool DoesProgramExist(string name, string args, Predicate<Process> pred) {
+        try {
+            ProcessStartInfo start = new ProcessStartInfo {
+                FileName = name,
+                Arguments = args,
+                UseShellExecute = false,
```

Comment:
this is the default value

Reply:
I blame tML for this, as I used it as reference for how to exec stuff :)